dashboard/src/components/AppUpgrade/AppUpgrade.v2.tsx

```diff
@@ -88,12 +88,21 @@ function AppUpgrade({
       chart.metadata &&
       chart.metadata &&
       chart.metadata.name &&
-      chart.metadata.version
+      chart.metadata.version &&
+      !deployed.chartVersion
     ) {
       const chartID = `${repoName}/${chart.metadata.name}`;
       getDeployedChartVersion(cluster, repoNamespace, chartID, chart.metadata.version);
     }
```

Comment:
This is my bet to try to remove the flakiness in the `private-registry` test that fails sometimes because it re-select the old version when upgrading. I have verified that the selector shows the latest version before going to the "Changes" tab so I am assuming this function (or the one in the UpgradeForm) is called twice with the old version, removing the changes done after selecting the newer version :crossed_fingers:

Reply:
example: https://circle-production-customer-artifacts.s3.amazonaws.com/picard/59ee02fec9e77c0001448c0b/5f636ba76982ad3245d65690-0-build/artifacts/integration/reports/screenshots/creates-a-private-registry.png?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20200917T143629Z&X-Amz-SignedHeaders=host&X-Amz-Expires=60&X-Amz-Credential=AKIAJR3Q6CR467H7Z55A%2F20200917%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Signature=18081a8d550269010165492aa6afae502fd3552fef500fd6ec666f3e70de8cbf

Reply:
_flips the table_
https://app.circleci.com/pipelines/github/kubeapps/kubeapps/1106/workflows/56a80055-e3b5-4814-aac9-6830db9bd795/jobs/29987/steps

Reply:
Painful :/ We should probably remove that change if it's not having the expected effect (avoid confusion later). Also, if this was related, we'd be better to reproduce the issue in a unit-test (much easier to test if it is the actual issue, but also documents the issue)?

Reply:
I have no idea what can be the issue, added a small sleep to at least avoid the flakiness (500ms is enough to avoid the issue, apparently)